src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -204,11 +208,57 @@ SDCPUThread::onBeginFetch()
     if (!_cpuPtr->requestInstAddrTranslation(fetch_req, getThreadContext(),
         [this](Fault f, const RequestPtr& r) {onPCTranslated(f, r);})) {
         panic("The CPU rejected my fetch translate request and I haven't been "
-            "programmed to know how to continue!!!");
+              "programmed to know how to continue!!!");
         // TODO reschedule a fetch for the future?
     }
 }
 
+
+void
+SDCPUThread::onDataAddrTranslated(Fault fault, const RequestPtr& req,
+                                  InflightInst& inst, bool write)
+{
+    if (fault != NoFault) {
+        // TODO handle fault.
+        panic("SDCPU not able to handle data translation faults yet!");
+    }
+
+    // TODO set the effective address
+    // TODO add dependencies according to the now known effective address, and
+    //      unset this instruction ready if appropriate.
+
+    if (write) {
+        if (!_cpuPtr->requestMemWrite(req, getThreadContext(),
+                                      [this, &inst](Fault fault) {
+                                          onExecutionCompleted(fault, inst);
+                                      })
+        ) {
+            panic("The CPU rejected my mem access request and I havne't been "
+                  "programmed to know how to continue!!!");
+        }
+    } else { // read
+        if (!_cpuPtr->requestMemRead(req, getThreadContext(),
+                                     [this, &inst](Fault fault) {
+                                         onExecutionCompleted(fault, inst);
+                                     })
+        ) {
+            panic("The CPU rejected my mem access request and I havne't been "
+                  "programmed to know how to continue!!!");
+        }
+    }
+}
+
+void
+SDCPUThread::onExecutionCompleted(Fault fault, InflightInst& inst)
+{
+    // The InflightInst will also match any of its dependencies automatically,
+    // and perform any ready callbacks if any instructions are made ready as a
+    // result of this instruction completed
+    inst.notifyComplete();
```

Comment:
I think you need to call static_inst->completeAcc()?

Reply:
I was intending for the CPU to make that call prior to the onExecutionCompleted call, since I'm basically letting the CPU distinguish between memory request writebacks and regular writebacks.

In this code, I'm treating both StaticInst::execute() and StaticInst::completeAcc() as some ISA-defined set of tasks that will produce accesses to the ExecContext, after which the instruction is considered Complete.